src/post_setup.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+sed -i '/^users:/ {N; s/users:.*/users: []/g}' /etc/cloud/cloud.cfg
+rm --force /etc/sudoers.d/90-cloud-init-users
+rm --force /root/.ssh/authorized_keys
+/usr/sbin/userdel --remove --force ubuntu
+sudo snap switch --channel=candidate amazon-ssm-agent
+sudo snap refresh amazon-ssm-agent
+[[ -d /etc/amazon/ssm ]] || sudo mkdir -p /etc/amazon/ssm
```

Comment:
Prefer the log form of CLI options:
```suggestion
[[ -d /etc/amazon/ssm ]] || sudo mkdir --parents /etc/amazon/ssm
```